src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs

```diff
@@ -311,44 +311,47 @@
    (fn [{:keys [props state this]}]
      (let [{:keys [server-response show-create-dialog?]} @state
            {:keys [clusters server-error]} server-response]
-        [:div {:display "inline-flex"}
-         (when show-create-dialog?
-           [ClusterCreator (assoc props :dismiss #(swap! state dissoc :show-create-dialog?)
-                                        :reload-after-create #(this :-get-clusters-list-if-whitelisted))])
-         [:div {} [:span {:data-test-id "spark-clusters-title" :style {:fontSize "125%" :fontWeight 500 :paddingBottom 10}} "Spark Clusters"]]
-         (if server-error
-           [comps/ErrorViewer {:error server-error}]
-           (if clusters
-             [NotebooksTable
-              (assoc props :toolbar-items [flex/spring [buttons/Button {:text "Create Cluster..." :style {:marginRight 7}
-                                                                        :data-test-id "create-modal-button"
-                                                                        :onClick #(swap! state assoc :show-create-dialog? true)}]]
-                           :clusters clusters
-                           :reload-after-delete #(this :-get-clusters-list-if-whitelisted))]))]))
+       [:div {:display "inline-flex"}
+        (when show-create-dialog?
+          [ClusterCreator (assoc props :dismiss #(swap! state dissoc :show-create-dialog?)
+                                       :reload-after-create #(this :-get-clusters-list-if-whitelisted))])
+        [:div {} [:span {:data-test-id "spark-clusters-title" :style {:fontSize "125%" :fontWeight 500 :paddingBottom 10}} "Spark Clusters"]]
+        (if server-error
+          [comps/ErrorViewer {:error server-error}]
+          (if clusters
+            [NotebooksTable
+             (assoc props :toolbar-items [flex/spring [buttons/Button {:text "Create Cluster..." :style {:marginRight 7}
+                                                                       :data-test-id "create-modal-button"
+                                                                       :onClick #(swap! state assoc :show-create-dialog? true)}]]
+                          :clusters clusters
+                          :reload-after-delete #(this :-get-clusters-list-if-whitelisted))]))]))
    :component-did-mount
    (fn [{:keys [this]}]
      (this :-is-leo-whitelisted))
+   :component-will-unmount
+   (fn [{:keys [locals]}]
+     (swap! locals assoc :dead? true))
```

Comment:
I thought unmounting took place immediately before a component is destroyed. How does the value of `:dead?` have an effect in this state?

Reply:
The component is unmounted, but the JS hangs around.